### PR TITLE
`PurchaseTester`: using new `StoreProduct.introductoryPrice` in SwiftPaywall

### PR DESCRIPTION
**Pull request by NachoSoto**
#848 and [sc-11623].
Depends on #1038.

Also documented some more methods in `Package` (#1046).

### Questions:
- `Package.localizedIntroductoryPriceString` used to be non-optional and default to an empty string (thank you `SwiftAPITester` for helping me catch this!).
Making it optional would be a breaking change, but I think 4.0 would be a good chance to do that. Version 3.0 doc just said _“A String containing the localized introductory price”_ with no indication that it could return `“”`. Or better yet, if there was no `introductoryPrice`, I guess it would return `$0.00`?

### CODE REVIEW

**Review by taquitos — COMMENTED** [review]
This looks good to me, but I'll wait for somebody who has more experience with this piece to 👍

File: Purchases/Public/Package.swift [hunk 1]
```diff
@@ -74,18 +74,22 @@ private extension PackageType {
...
-    @objc public var localizedIntroductoryPriceString: String {
-        if #available(iOS 11.2, macOS 10.13.2, tvOS 11.2, *) {
-            // todo: uncomment when product discounts are supported in product details
-            // return storeProduct.localizedIntroductoryPriceString
-            return ""
-        } else {
-            return ""
+    /// The price of the `StoreProduct.introductoryPrice` formatted using ``priceFormatter``.
+    /// - Returns: `nil` if there is no `introductoryPrice`.
+    @objc public var localizedIntroductoryPriceString: String? {
```
Comment (aboedo) [inline Purchases/Public/Package.swift:84]: [not needed for this PR] I wonder if this is worth having as a property of StoreProduct and bridging it over here

**Review by aboedo — APPROVED** [review]
looks great! 🎉

File: Purchases/Public/Package.swift (hunk 1 above)
Comment (NachoSoto) [reply Purchases/Public/Package.swift:84]: Yeah I went back and forth. I guess it makes sense to have all these computed properties in StoreProduct instead 👍

Comment (NachoSoto) [reply Purchases/Public/Package.swift:84]: Done.

**Pull request merged**